client/header/activity-panel/index.js

```diff
@@ -254,4 +257,36 @@ class ActivityPanel extends Component {
 	}
 }
 
-export default clickOutside( ActivityPanel );
+export default withSelect( select => {
+	const { getReportItems, getReportItemsError, isReportItemsRequesting } = select( 'wc-api' );
+	const orderStatuses = wcSettings.wcAdminSettings.woocommerce_actionable_order_statuses || [
+		'processing',
+		'on-hold',
+	];
+
+	if ( ! orderStatuses.length ) {
+		return { unreadOrders: false };
+	}
+
+	const ordersQuery = {
+		page: 1,
+		per_page: 0,
+		status_is: orderStatuses,
+	};
+
+	const totalOrders = getReportItems( 'orders', ordersQuery ).totalResults;
+	const isError = Boolean( getReportItemsError( 'orders', ordersQuery ) );
+	const isRequesting = isReportItemsRequesting( 'orders', ordersQuery );
+
+	let unreadOrders = null;
```

Comment:
The problem is that when `isRequesting` or `isError` are true, we want `unreadOrders` to be `null` instead of false. That's because this value is later used to set `isEmpty` in `<OrdersPanel>`, and we don't want to set it while loading or if there was an error, because it prevents `OrdersPanel` to request data. Does that make sense?